### ISSUE TITLE
aws_elastic_beanstalk_hosted_zone Unsupported region: eu-north-1

### ISSUE THREAD

**Issue by Vadim-Zenin** [terraform-providers/terraform-provider-aws#7763]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

`terraform -v
Terraform v0.11.11`
### Affected Resource(s)

* aws_XXXXX
* AWS Beanstalk with Application load balancer environment

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp

resource "aws_elastic_beanstalk_environment" "default" {
  name                = "${var.website_name}-${var.website_type}-${var.environment}-ebs-env"
  application         = "${aws_elastic_beanstalk_application.website.name}"
  tier                = "${var.tier}"
  solution_stack_name = "${data.aws_elastic_beanstalk_solution_stack.single_docker.name}"
  cname_prefix = "${var.website_name}-${var.website_type}-${var.environment}"
  version_label = "${aws_elastic_beanstalk_application_version.website.name}"
  setting {
    namespace = "aws:autoscaling:launchconfiguration"
    name      = "InstanceType"
    value     = "${var.ebs_instance_type}"
  }
  setting {
    namespace = "aws:autoscaling:launchconfiguration"
    name      = "IamInstanceProfile"
    value     = "${var.ebs_iam_instance_profile}"
  }
  setting {
    namespace = "aws:autoscaling:asg"
    name      = "MaxSize"
    value     = "1"
  }
  setting {
    name = "EnvironmentType"
    namespace = "aws:elasticbeanstalk:environment"
    value = "LoadBalanced"
  }
  setting {
    name = "LoadBalancerType"
    namespace = "aws:elasticbeanstalk:environment"
    value = "application"
  }
  setting {
    namespace = "aws:elbv2:loadbalancer"
    name      = "SecurityGroups"
    value     = "${module.aws_ssm_parameter_store.ssmp_value}"
  }
  setting {
    namespace = "aws:elbv2:listener:default"
    name      = "ListenerEnabled"
    value     = "true"
  }
  setting {
    namespace = "aws:elbv2:listener:443"
    name      = "ListenerEnabled"
    value     = "true"
  }
  setting {
    namespace = "aws:elbv2:listener:443"
    name      = "Protocol"
    value     = "HTTPS"
  }
  setting {
    namespace = "aws:elbv2:listener:443"
    name      = "SSLCertificateArns"
    value     = "${module.aws_alb_certificate.acm_certificate_arn}"
  }
  setting {
    namespace = "aws:elbv2:listenerrule:default"
    name = "PathPatterns"
    value = "/*"
  }
  ###  Application Load Balancer Health check settings ###
  # The Application Load Balancer health check does not take into account the Elastic Beanstalk health check path
  # http://docs.aws.amazon.com/elasticbeanstalk/latest/dg/environments-cfg-applicationloadbalancer.html
  # http://docs.aws.amazon.com/elasticbeanstalk/latest/dg/environments-cfg-applicationloadbalancer.html#alb-default-process.config
  setting {
    namespace = "aws:elasticbeanstalk:environment:process:default"
    name      = "HealthCheckPath"
    value     = "${var.ebs_health_check_url}"
  }
  setting {
    namespace = "aws:elasticbeanstalk:environment:process:default"
    name      = "Port"
    value     = "80"
  }
  setting {
    namespace = "aws:elasticbeanstalk:environment:process:default"
    name      = "Protocol"
    value     = "HTTP"
  }
  setting {
    namespace = "aws:elasticbeanstalk:application"
    name      = "Application Healthcheck URL"
    value     = "${var.ebs_health_check_url}"
  }
  setting {
    namespace = "aws:elasticbeanstalk:command"
    name      = "DeploymentPolicy"
    value     = "Rolling"
  }
  setting {
    namespace = "aws:autoscaling:updatepolicy:rollingupdate"
    name      = "RollingUpdateEnabled"
    value     = "true"
  }
  setting {
    namespace = "aws:autoscaling:updatepolicy:rollingupdate"
    name      = "RollingUpdateType"
    value     = "Health"
  }

  tags {
    "website_id" = "${var.website_name}"
    "environment" = "${var.environment}"
    "Terraform-managed" = true
  }
}

data "aws_elastic_beanstalk_hosted_zone" "default" {
  depends_on  = ["aws_elastic_beanstalk_environment.default"]
}

output "ebs_hosted_zone_id" {
  description = "Beanstalk ALB zone id"
  value = "${data.aws_elastic_beanstalk_hosted_zone.default.id}"
}

```

### Panic Output

```
aws_elastic_beanstalk_environment.default: Still creating... (4m50s elapsed)
aws_elastic_beanstalk_environment.default: Creation complete after 4m59s (ID: e-111111111)
data.aws_elastic_beanstalk_hosted_zone.default: Refreshing state...

Error: Error applying plan:

1 error(s) occurred:

* data.aws_elastic_beanstalk_hosted_zone.default: data.aws_elastic_beanstalk_hosted_zone.default: Unsupported region: eu-north-1

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

### Expected Behavior

Provide output with Beanstalk's Application load balancer zone ID.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

It works in different regions (ca-central-1 for example)
### References

* https://github.com/terraform-providers/terraform-provider-aws/issues/2952

**Comment by bflad** [comment]
Hi @Vadim-Zenin 👋 Thanks for submitting this. We generate the data source information based on the [AWS Regions and Endpoints documentation](https://docs.aws.amazon.com/general/latest/gr/rande.html#elasticbeanstalk_region), which oddly is still missing the value.

Are you aware if there is a correct value and what it might be? Thanks.

**Comment by Vadim-Zenin** [comment]
Hi @bflad ,
Thank you for answer.
I got answer from AWS supports and documentation teams.
eu-north-1 Route 53 Hosted Zone ID added to page https://docs.aws.amazon.com/en_us/general/latest/gr/rande.html#elasticbeanstalk_region .

**Comment by Vadim-Zenin** [comment]
The issue exists still:
```
Error: Error applying plan:
1 error(s) occurred:
* data.aws_elastic_beanstalk_hosted_zone.default: data.aws_elastic_beanstalk_hosted_zone.default: Unsupported region: eu-north-1
```

**Comment by bflad** [comment]
@Vadim-Zenin sorry I probably should not have used the word "generate" before; these values are currently hardcoded in the data source. This pull request will update the data source: #7829

**Comment by bflad** [comment]
Support for the new region has been merged in and will release with version 2.1.0 of the Terraform AWS Provider, likely in the next day or two.

**Comment by Vadim-Zenin** [comment]
@bflad thank you.

**Comment by bflad** [comment]
This has been released in [version 2.1.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#210-march-07-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
